modules/mongodb-atlas/users.tf

```diff
@@ -20,7 +20,7 @@ resource "mongodbatlas_database_user" "user" {
   }
 
   scopes {
-    name   = "cluster"
+    name   = var.cluster_name
```

Comment:
Update to point to var.project_name instead as that is what becomes cluster_name.
See: https://github.com/dasmeta/terraform-aws-modules/blob/d2346e20a1424a16ede9e3dfd72d856a18e98292/modules/mongodb-atlas/cluster.tf#L3